packages/sd-jwt/src/action-handler.ts

```diff
@@ -225,10 +225,29 @@ export class SDJwtPlugin implements IAgentPlugin {
     if (!payload.cnf) {
       throw Error('other method than cnf is not supported yet')
     }
-    const key = payload.cnf.jwk as JsonWebKey
-    return this.verifySignatureCallback(context)(data, signature, key)
+    return this.verifySignatureCallback(context)(data, signature, this.getJwk(payload))
   }
 
+  private getJwk(payload: JwtPayload): JsonWebKey {
+    if (payload.cnf?.jwk !== undefined) {
+      return payload.cnf.jwk as JsonWebKey
+    } else if (payload.cnf !== undefined && 'kid' in payload.cnf && typeof payload.cnf.kid === 'string' && payload.cnf.kid.startsWith('did:jwk:')) {
+      // extract JWK from kid FIXME isn't there a did function for this already? Otherwise create one
+      const encoded = this.extractBase64FromDID(payload.cnf.kid)
```

Comment:
We already have low level support converting any VM to a JWK. So this method could also deal with more than did:jwk: as that is way too naive anyway. The chances of a CNF being a did:jwk only are pretty slim

Reply:
https://github.com/Sphereon-Opensource/SSI-SDK-crypto-extensions/blob/c0ca69fe0f10cfd9cdafa94b7af31a6cf6100680/packages/did-utils/src/did-functions.ts#L293C4-L294C1

Reply:
https://github.com/Sphereon-Opensource/SSI-SDK-crypto-extensions/blob/c0ca69fe0f10cfd9cdafa94b7af31a6cf6100680/packages/did-utils/src/did-functions.ts#L451

Reply:
And many other functions you can use in that package. When dealing with DIDs look into the did-utils module. When dealing with key conversions look into the key-utils module. Code like the above should never live in the SD-JWT module